src/TortoiseProc/CommitDlg.cpp

```diff
@@ -1478,8 +1524,8 @@ UINT CCommitDlg::StatusThread()
 				GitRev headRevision;
 				if (headRevision.GetParentFromHash(hash))
 					MessageBox(headRevision.GetLastErr(), L"TortoiseGit", MB_ICONERROR);
-				// do not allow to show diff to "last" revision if it has more that one parent
-				if (headRevision.ParentsCount() != 1)
+				// do not allow to show diff to "last" revision if it has more that one parent or staging support is enabled
+				if (headRevision.ParentsCount() != 1 || m_bStagingSupport)

```

Comment:
* Why is this needed?
* "Diff to last commit" is not automatically unchecked when staging support is activated.

Reply:
> "Diff to last commit" is not automatically unchecked when staging support is activated.

That's right. Only when doing an amend, "Show diff to last commit" must stay checked and grayed out.

> Why is this needed?

Comment clarified.